### PR TITLE
Rework nick after death

### PR DESCRIPTION
**Pull request by rexim**

### CODE REVIEW

File: pacmacs.el [hunk 1]
```diff
@@ -80,6 +81,8 @@
 
 (defvar pacmacs-waiting-counter 0)
 
+(defvar pacmacs-game-over-state-rendered nil)
```
Comment (rexim) [inline pacmacs.el]: It's not needed anymore. Remove that.

File: pacmacs.el [hunk 2]
```diff
@@ -91,6 +94,9 @@
   (setq cursor-type nil)
   (setq truncate-lines t))
 
+(define-derived-mode pacmacs-game-over-mode text-mode "pacmacs-game-over-mode"
```
Comment (rexim) [inline pacmacs.el]: Not sure if it's needed too

**Comment by ForNeVeR** [comment]
Well, looks good to me. Please fix the unit tests and make changes you mentioned in the comments. You're free to merge it afterwards.

**Comment by rexim** [comment]
Thank you mr @ForNeVeR! You're the best! :)

**Comment by ForNeVeR** [comment]
It seems that you're changing much more that I expected. Please notify me if you need my help to review this again after all these changes.

**Comment by rexim** [comment]
@ForNeVeR yep, that's true. I just wanted to slightly refactor that mess ~~and ended up removing everything~~. I really need to get this reviewed after I finish everything. So, once I'm done I will reassign this to you for an additional review. Thanks!

**Comment by rexim** [comment]
@ForNeVeR, Sending back for an additional review. Thanks!

File: test/pacmacs-score-test.el [hunk 3]
```diff
@@ -15,13 +15,21 @@
...
                        ("world" . 20)
                        ("bar" . 10)))
-        (expected-string (concat "Best Scores:\n"
-                                 "------------\n"
-                                 "foo   40\n"
-                                 "hello 30\n"
-                                 "world 20\n"
-                                 "bar   10\n")))
+        (expected-string (concat ;"Best Scores:\n"
+                                 ;"------------\n"
```
Comment (ForNeVeR) [inline test/pacmacs-score-test.el]: Why are these lines commented out?

File: test/pacmacs-test.el [hunk 4]
```diff
@@ -126,3 +126,34 @@
 
      (should (null pacmacs--ghosts))
      (should (= 11 (length pacmacs--terrified-ghosts))))))
+
+(ert-deftest pacmacs--align-score-record-nickname-test ()
+  (let ((pacmacs--max-score-nick-size 5)
```
Comment (ForNeVeR) [inline test/pacmacs-test.el:131]: Is it okay to change the constant value in such manner inside a unit test? I wasn't aware that elisp bindings are dynamic by default.

Comment (rexim) [inline test/pacmacs-test.el:131]: Yep, just wanted to show that the function also depends on a global constant. And yes, elisp bindings are always dynamic. And in my opinion that was the main reason why this dialect of lisp sucked. But recently they introduced lexical-binding mode, which makes let bindings dynamic only for global variables. Which is bit better.

File: test/pacmacs-score-test.el (hunk 3 above)
Comment (rexim) [inline test/pacmacs-score-test.el]: Total forgot about them. Thanks! I'll remove that.

File: test/pacmacs-test.el (hunk 4 above)
Comment (ForNeVeR) [inline test/pacmacs-test.el:131]: Well, okay then, let it be so.

**Comment by ForNeVeR** [comment]
I've finished with the review, it looks good.

**Comment by rexim** [comment]
More tanks for you mr @ForNeVeR :3

**Pull request merged**